common/lib/xmodule/xmodule/js/spec/html/edit_spec.js

```diff
@@ -6,26 +6,45 @@ describe('HTMLEditingDescriptor', function() {
       loadFixtures('html-edit-visual.html');
       this.descriptor = new HTMLEditingDescriptor($('.test-component'));
     });
-    it('Returns data from Visual Editor if text has changed', function() {
+    it('Returns data from Visual Editor if text has changed', function(done) {
       const visualEditorStub =
         {getContent() { return 'from visual editor'; }};
       spyOn(this.descriptor, 'getVisualEditor').and.callFake(() => visualEditorStub);
-      const { data } = this.descriptor.save();
-      expect(data).toEqual('from visual editor');
+
+      var self = this;
```

Comment:
@tecoholic, please add a quick explanation of why we're doing this.

Reply:
@Agrendalath Done.